lib/packwerk/association_inspector.rb

```diff
@@ -46,20 +60,42 @@ def constant_name_from_node(node, ancestors:)
       end
     end
 
-    private
+    sig { params(node: AST::Node).returns(T.nilable(String)) }
+    def active_model_serializers_constant_name(node)
+      arguments = NodeHelpers.method_arguments(node)
+      return unless (association_name = association_name(arguments))
+
+      if (class_name_node = has_value(arguments, key: :serializer))
+        return unless NodeHelpers.constant?(class_name_node)
+
+        T.cast(NodeHelpers.literal_value(class_name_node), String)
+      else
+        "#{@inflector.classify(association_name.to_s)}Serializer"
+      end
+    end
+
+    sig { params(node: AST::Node).returns(T.nilable(Symbol)) }
+    def type(node)
+      case NodeHelpers.source_location(node).match(%r{\bapp/(.*)/}).to_a.last
```

Comment:
If we don't filter paths somehow the bug still exists for AMS users. Maybe we can pull ignore paths from a configuration file if we think checking for `app/models` is too brittle?

Reply:
Can we check the context where `has_many` is called? This is tricky because you could make a case that serializers can also be inside `app/models`.

If we can't check the path, I'm ok with a exclude patter, so people that use AMS can say to not check on `app/serializers`

Reply:
So from my understanding, we can only check what is statically printed in the file. There's no way to guarantee a superclass if a model without actually running the code or making unsafe assumptions.

Checking the superclass of the context is `ActiveRecord::Base` or `ApplicationRecord` is too brittle (because it doesn't account for custom record subclasses), and serializers have the same problem with their immediate subclass. We could check if the superclass or class end with `Record`, but that is an unsafe assumption for non-abstract models. We could try to resolve superclasses that aren't Active Record default, but if any of them aren't autoloaded with Zeitwerk, we won't be able to see them.

I think an exclude glob option covers all of our bases and avoids taking on this complexity.

Reply:
Updated the PR to take this approach.